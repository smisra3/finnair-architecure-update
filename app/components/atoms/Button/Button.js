// @flow
import type { Node } from 'react';

import styled from 'styled-components';
import styles from './Button.style';
import Spinner from '../../atoms/Spinner';

type Props = {
  className: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  elementRef?: any => void,
  loader?: boolean,
  loaderPositionCenter?: boolean,
  disabled?: boolean,
  ariaLabel?: string,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  elementRef,
  loader,
  loaderPositionCenter,
  disabled,
  ariaLabel,
  ...others
}: Props): Node => (
  <button
    className={className}
    type={type}
    {...others}
    ref={elementRef}
    disabled={disabled || loader}
    aria-label={ariaLabel || null}
  >
    {loader && (
    <div className={`spinner ${loaderPositionCenter ? 'center-spinner' : ''}`}>
      <Spinner overlayEnabled spinner />
    </div>
      )}
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  elementRef: undefined,
  loader: false,
  disabled: false,
  loaderPositionCenter: false,
  ariaLabel: '',
};

export default styled(Button)`${styles}`;
export { Button as ButtonVanilla };
