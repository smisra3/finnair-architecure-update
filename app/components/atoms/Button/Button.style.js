import { css } from 'styled-components';

const ButtonStyles = css`
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  min-height: 48px;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? '0.3' : '1')};
  ${props =>
    (props.primary
      ? `
        background: ${props.theme.buttonBackgroundColorPrimary} ;
        color: ${props.theme.buttonColorPrimary};
        font-size: ${props.theme.fontSizeButtonPrimary};
        padding: 0 1rem;
      ` : '')};
  ${props =>
    (props.secondary
      ? `
        background: ${props.theme.buttonBackgroundColorSecondary} ;
        color: ${props.theme.buttonColorSecondary};
        font-size: ${props.theme.fontSizeButtonSecondary};
        padding: 0 1rem;
      ` : '')};
  ${props =>
    (props.fullWidth
      ? `
        width: 100% ;
      ` : '')};

  &:focus {
    outline: 1px dashed ${props => props.theme.colorOutline};
  }

  &:focus:not(.focus-visible) {
    outline: none;
  }

  .spinner {
    position: absolute;
    top: 50%;
    left: 9px;
    transform: translateY(-50%);

    &.center-spinner {
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
    }

    div {
      height: 25px;
      width: 25px;
      top: 0;
    }
  }
`;

export default ButtonStyles;
