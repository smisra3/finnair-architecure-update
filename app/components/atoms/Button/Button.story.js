// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  primaryButton,
  primaryButtonDisabled,
  secondaryButton,
  secondaryButtonDisabled,
} from './Button.mock';

// Import Vanilla Component to display markup and props
import { ButtonVanilla } from './Button';
// Import Styled Component to showcase variations
import Button from './index';

storiesOf('Atoms', module).addWithChapters('Buttons', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ButtonVanilla {...primaryButton} className={`hide-default-sample ${primaryButton.className}`}>Submit</ButtonVanilla>,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Button Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Button {...primaryButton}>Submit</Button>,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Button {...primaryButtonDisabled}>Submit</Button>,
        },
      ],
    },
    {
      title: 'Secondary Button Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Button {...secondaryButton}>Submit</Button>,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Button {...secondaryButtonDisabled}>Submit</Button>,
        },
      ],
    },
  ],
});
