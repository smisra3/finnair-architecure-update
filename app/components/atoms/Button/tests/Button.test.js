import { shallow } from 'enzyme';
import { ButtonVanilla } from '../Button';

describe('Button Component', () => {
  let button = '';
  beforeEach(() => {
    button = shallow(<ButtonVanilla>Submit</ButtonVanilla>);
  });

  test('should render correctly', () => {
    expect(button).toMatchSnapshot();
  });

  test('should render disabled', () => {
    button = shallow(<ButtonVanilla disabled>Submit</ButtonVanilla>);
    expect(button.find({ disabled: true })).toHaveLength(1);
  });

  test('should render title element', () => {
    button = shallow(<ButtonVanilla title="tooltip">Submit</ButtonVanilla>);
    expect(button.find({ title: 'tooltip' })).toHaveLength(1);
  });
});
