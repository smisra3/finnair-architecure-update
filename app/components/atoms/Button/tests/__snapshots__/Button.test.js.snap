// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<button
  aria-label={null}
  disabled={false}
  type="button"
>
  Submit
</button>
`;
